src/config.rs

```diff
@@ -482,10 +494,15 @@ impl Account {
             && self.token_uri == act_dump.token_uri
     }
 
-    pub fn redirect_uri(&self, http_port: u16) -> Result<Url, Box<dyn Error>> {
+    pub fn redirect_uri(&self, http_port: u16, https_port: u16) -> Result<Url, Box<dyn Error>> {
         let mut url = Url::parse(&self.redirect_uri)?;
-        url.set_port(Some(http_port))
-            .map_err(|_| "Cannot set port")?;
+        if self.redirect_uri.starts_with("https") {
```

Comment:
One problem I can see here is that the scheme is case insensitive so a user (or host) could send us `HtTpS` (or whatever) and we wouldn't match correctly here. We could call `to_lowercase` here, or perhaps earlier in the system, or perhaps make `redirect_uri` an enum or similar.